src/modules/advertisement/advertisement.controller.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  ForbiddenException,
+  Get,
+  Param,
+  Patch,
+  Post,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common'
+import { FileInterceptor } from '@nestjs/platform-express'
+import { ApiOperation } from '@nestjs/swagger'
+
+import { Auth } from '~/common/decorator/auth.decorator'
+import { ApiName } from '~/common/decorator/openapi.decorator'
+
+import { UploadService } from '../upload/upload.service'
+import { goUrlDto } from './advertisement.dto'
+import { AdvertisementService } from './advertisement.service'
+
+@Controller('advertisement')
+@ApiName
+export class AdvertisementController {
+  constructor(
+    private readonly advertisementService: AdvertisementService,
+    private readonly uploadService: UploadService,
+  ) {}
+
+  @Get()
+  @ApiOperation({ summary: '广告查找' })
+  async findUrl() {
+    const result = await this.advertisementService.findUrl()
+    return result
+  }
+
+  @Post()
+  @Auth()
```

Comment:
```suggestion
  @Admin()
  @Auth()
```
这里其它几个跟鉴权有关的接口都加下，表示只有管理员才可以设置